gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/integration/use_case/IngestIntegrationApisUseCase.java

```diff
@@ -90,50 +96,116 @@ public Completable execute(Input input) {
                         }
                     });
             })
             .ignoreElements();
     }
 
-    private void createDocumentation(IntegrationApi integrationApi, String referenceId, AuditInfo auditInfo) {
-        integrationApi
-            .pages()
-            .stream()
-            .flatMap(page -> {
-                if (page == null || page.pageType() == null) {
-                    return Stream.empty();
-                }
-                return switch (page.pageType()) {
-                    case SWAGGER -> Stream.of(buildSwaggerPage(integrationApi.name(), referenceId, page.content()));
-                    case ASYNCAPI -> Stream.of(buildAsyncApiPage(integrationApi.name(), referenceId, page.content()));
-                    case ASCIIDOC, MARKDOWN, MARKDOWN_TEMPLATE -> {
-                        log.error("Impossible to import {} documentation for {}", page.pageType(), integrationApi.name());
-                        yield Stream.empty();
-                    }
-                };
-            })
-            .forEach(page -> createApiDocumentationDomainService.createPage(page, auditInfo));
+    private void createApi(Api federatedApi, IntegrationApi integrationApi, PrimaryOwnerEntity primaryOwner, AuditInfo auditInfo) {
+        try {
+            createApiDomainService.create(federatedApi, primaryOwner, auditInfo, validateFederatedApi::validateAndSanitizeForCreation);
+
+            Optional
+                .ofNullable(integrationApi.plans())
+                .ifPresent(plans ->
+                    plans
+                        .stream()
+                        .map(p -> PlanModelFactory.fromIntegration(p, federatedApi))
+                        .forEach(p -> createPlanDomainService.create(p, List.of(), federatedApi, auditInfo))
+                );
+
+            Optional
+                .ofNullable(integrationApi.pages())
+                .ifPresent(pages ->
+                    pages
+                        .stream()
+                        .flatMap(page -> buildPage(page, integrationApi, federatedApi.getId()))
+                        .forEach(page -> createApiDocumentationDomainService.createPage(page, auditInfo))
+                );
+        } catch (Exception e) {
+            log.warn("An error occurred while importing api {}", federatedApi, e);
+        }
+    }
+
+    private void updateApi(Api federatedApi, IntegrationApi integrationApi, AuditInfo auditInfo, PrimaryOwnerEntity primaryOwner) {
+        log.debug("API already ingested [id={}] [name={}], performing update", federatedApi.getId(), federatedApi.getName());
+        try {
+            /*
+             * Because we use page names as identifiers we need to look for the page by existing API name.
+             * Otherwise, in case of an API name update, we would create a new page instead of updating the existing one.
+             * If we move on to some predictable unique ID the existingApiName will become unnecessary.
+             */
```

Comment:
I think Jonathan agreed to take care of that later so I created this Jira issue to not forget about it in the future:
https://gravitee.atlassian.net/browse/APIM-5881 

Unless you think it might be better to deal with that now than later